### PR TITLE
refactor: handle intent orders in fetch bridge tx function

### PR DESCRIPTION
**Pull request by oscarwroche**
## Explanation

Removed the dedicated intent polling path and folded intent updates into the existing bridge status polling flow. Intent status translation now lives in `intent-api` to align with `fetchBridgeTxStatus` responses, with shared handling for completion, failure, and transaction sync. Updated and added tests to cover the new translation and intent polling behavior.

## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've communicated my changes to consumers by [updating changelogs for packages I've changed](https://github.com/MetaMask/core/tree/main/docs/processes/updating-changelogs.md)
- [ ] I've introduced [breaking changes](https://github.com/MetaMask/core/tree/main/docs/processes/breaking-changes.md) in this PR and have prepared draft pull requests for clients and consumer packages to resolve them

---

> [!NOTE]
> **Medium Risk**
> Refactors core polling/status update logic and changes intent history keying (dropping the `intent:` prefix), which could impact persisted `txHistory` lookups and TransactionController syncing if any edge cases were missed. Risk is mitigated by expanded unit test coverage around intent/bridge polling, approval flows, and failure handling.
> 
> **Overview**
> Unifies intent-order status polling with the existing `#fetchBridgeTxStatus` flow by detecting intent quotes and querying `IntentApiImpl.getOrderStatus`, removing the old dedicated intent polling path.
> 
> Adds `translateIntentOrderToBridgeStatus` (and `IntentStatusTranslation`) in `utils/intent-api` plus a new `IntentStatusManager` to sync intent status into TransactionController without clobbering required `TransactionMeta` fields.
> 
> Changes intent history storage to use the raw `orderUid` as the `txHistory` key (instead of `intent:${id}`) and removes `BridgeHistoryItem.srcTxHashes`; extracts shared helpers (`waitForTxConfirmation`, `rekeyHistoryItemInState`) into `utils/transaction`.
> 
> Updates/extends tests accordingly (including approval-confirmation branches, missing-history early returns, Tron client request options, and swap approval batching) and raises Jest coverage thresholds to 100% lines/statements.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8ff31ec983141147fc30ac05cf885277a6cc8d08. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/bridge-status-controller/src/__snapshots__/bridge-status-controller.test.ts.snap [hunk 1]
```diff
@@ -114,6 +114,9 @@ Object {
       ],
     },
     "slippagePercentage": 0,
+    "srcTxHashes": Array [
+      "0xsrcTxHash1",
+    ],
```
Comment (micaelae) [inline packages/bridge-status-controller/src/__snapshots__/bridge-status-controller.test.ts.snap]: I don't see `srcTxHashes` being used in the mobile codebase. And the only usage I see in the controller is for updating the history item. Can this field be removed?

File: packages/bridge-status-controller/src/utils/bridge-status-controller-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      }
+    }
+
+    if (Date.now() - start > timeoutMs) {
+      throw new Error('Timed out waiting for approval confirmation');
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, pollMs));
+  }
+};
```
Comment (micaelae) [inline packages/bridge-status-controller/src/utils/bridge-status-controller-helpers.ts]: Nice! Can you move this to transactions.ts? that's where the TransactionController-related utils are

File: packages/bridge-status-controller/src/bridge-status-controller.ts [hunk 3]
```diff
@@ -1853,13 +1662,13 @@ export class BridgeStatusController extends StaticIntervalPollingController<Brid
 
       // Record in bridge history with actual transaction metadata
       try {
-        // Use 'intent:' prefix for intent transactions
-        const bridgeHistoryKey = `intent:${orderUid}`;
+        // Use orderId as the history key for intent transactions
+        const bridgeHistoryKey = orderUid;
```
Comment (micaelae) [inline packages/bridge-status-controller/src/bridge-status-controller.ts:1654]: This is a breaking change btw - existing intent txHistory items will need to be migrated to use the new key format

File: packages/bridge-status-controller/src/bridge-status-controller.ts [hunk 4]
```diff
@@ -676,42 +664,75 @@ export class BridgeStatusController extends StaticIntervalPollingController<Brid
...
+        const intentApi = new IntentApiImpl(
+          this.#config.customBridgeApiBaseUrl,
+          this.#fetchFn,
+        );
+        const intentOrder = await intentApi.getOrderStatus(
+          bridgeTxMetaId,
+          historyItem.quote.intent?.protocol ?? '',
+          srcChainId.toString(),
+          this.#clientId,
+        );
```
[the remark this answers is not in the document]
Comment (oscarwroche) [reply packages/bridge-status-controller/src/bridge-status-controller.ts:705]: Migration here https://github.com/MetaMask/metamask-mobile/pull/25525

File: packages/bridge-status-controller/src/bridge-status-controller.ts (hunk 3 above)
Comment (oscarwroche) [reply packages/bridge-status-controller/src/bridge-status-controller.ts:1654]: https://github.com/MetaMask/metamask-mobile/pull/25525

File: packages/bridge-status-controller/src/bridge-status-controller.intent.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    updateTransactionFn,
+  }: {
+    messenger: BridgeStatusControllerMessenger;
+    updateTransactionFn: typeof TransactionController.prototype.updateTransaction;
+  }) {
+    this.#messenger = messenger;
+    this.#updateTransactionFn = updateTransactionFn;
+  }
+
+  syncTransactionFromIntentStatus = (
```
Comment (micaelae) [inline packages/bridge-status-controller/src/bridge-status-controller.intent.ts:25]: Creating a new class for syncing the status seems unnecessary here. Can you either move this to the intent class or write it as a util (similar to `findAndUpdateTransactionsInBatch`)?

File: packages/bridge-status-controller/src/bridge-status-controller.intent.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    bridgeTxMetaId: string,
+    historyItem: BridgeHistoryItem,
+    intentTranslation: IntentStatusTranslation,
+    intentOrderStatus: string,
```
Comment (micaelae) [inline packages/bridge-status-controller/src/bridge-status-controller.intent.ts:29]:
```suggestion
    intentOrderStatus: IntentOrderStatus,
```

File: packages/bridge-status-controller/src/bridge-status-controller.intent.ts (hunk 5 above)
Comment (oscarwroche) [reply packages/bridge-status-controller/src/bridge-status-controller.intent.ts:25]: @micaelae I can change it back to a util as the class is not strictly necessary but I'm planning on moving more stuff to this class later. The goal of the PR is to clearly separate the intent logic from the rest to make the migration of additional intent logic to this controller easier.

**Review by micaelae — APPROVED** [review]
